python/doc/source/other_systems.rst

```diff
@@ -0,0 +1,158 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. currentmodule:: pyarrow
+.. _other_systems:
+
+Using Arrow with other systems
+==============================
+
+Timestamps
```

Comment:
I don't know the correct term for these but I think there are two aspect of a timestamp:
1. What's the numeric value and what it represents:
This is important because when passing the numeric value between different system, if they mean different things, then we have an issue. The key definition is `UTC-normalized` and `non-UTC-normalized`.
For instance, in Arrow, the numeric value is always normalized to UTC, and in Spark, it's not always the case. The numeric defines the point in time of the timestamp.

2. How is the timestamp interpreted:
This is related to timezone. Given a numeric value and its point in time,  the timezone information decides a timestamp is interpreted .
For instance, in Arrow, this is the `timeZone` in the meta data. In python, it's tz_convert, in Spark, it's changing System timezone or `SESSION_LOCAL_TIMEZONE`. These don't change the numeric value of a timestamp, but changes how they are interpreted.

Reply:
I think have good definition of these and clarifying first would help user understanding timestamp presentation and operation in different systems.

Reply:
Thanks Lee, I will add an explanation of this sort.